### PR TITLE
Make all submission tables respects the SUBMISSIONS_TABLE_EXCLUDED_FIELDS setting

### LINKED ISSUE #3758: All tables that make use of BaseAdminSubmissionsTable does not take SUBMISSIONS_TABLE_EXCLUDED_FIELDS in to account
An example is the "Submissions waiting for your review " table on staffs dashboard. It displaces fields that are set to be excluded in the `SUBMISSIONS_TABLE_EXCLUDED_FIELDS` setting.

### PR DESCRIPTION
**Pull request by frjo**
Fixes #3758

## Test Steps

 - [ ] Check that tables on the dashboard do not contain columns that have been excluded.

### CODE REVIEW

**Comment by wes-otf** [comment]
@frjo is this deployed? On the test instance I still see `Organization name` under `Your previous reviews` despite `SUBMISSIONS_TABLE_EXCLUDED_FIELDS=meta_terms,organization_name` in heroku but I could be misunderstanding what the target is here:
Screenshot 2024-04-17 at 10 10 10

Screenshot 2024-04-17 at 10 10 27

**Comment by frjo** [comment]
Seems to be a bug in my implementation. Does not have any effect on the views "…waiting for your review/…previous reviews".

But it does work on e.g. "Staff flagged submissions"

@theskumar @sandeepsajan0 Any ideas here?

File: hypha/apply/funds/tables.py [hunk 1]
```diff
@@ -188,8 +191,16 @@ class Meta(SubmissionsTable.Meta):
...
             "reviews_stats",
             "organization_name",
-        )
-        sequence = fields + ("comments",)
+        ]
+
+        # Remove fields that have been added to the exclude fields settings.
+        fields = [
+            x for x in fields if x not in settings.SUBMISSIONS_TABLE_EXCLUDED_FIELDS
+        ]
```
Comment (sandeepsajan0) [inline hypha/apply/funds/tables.py]: It doesn't seem to be working that way. I looked and tried to update its base columns and sequence in `__init__` then it works.
here is the code:
```    
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

        excluded_columns = getattr(settings, "SUBMISSIONS_TABLE_EXCLUDED_FIELDS", [])
        for column_name in excluded_columns:
            if column_name in self.base_columns:
                del self.base_columns[column_name]
        self.sequence = [column_name for column_name in self._meta.fields if column_name not in excluded_columns]```

**Comment by sandeepsajan0** [comment]
Rebased and added the commented code. @frjo You may test it again, it should work as expected now.

File: hypha/apply/funds/tables.py (hunk 1 above)
Comment (sandeepsajan0) [reply hypha/apply/funds/tables.py]: I've added the code.

**Comment by frjo** [comment]
Closing this in favour of future direction where tables on dashboards are replaced with lists that show the info people has most use of.

**Pull request closed without merging**